src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -53,54 +55,64 @@ public void testDoGet() throws IOException, ServletException {
   }
 
   @Test
-  public void testDoPost_BadUsername() throws IOException, ServletException {
-    Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
+  public void testDoPost_UnregisteredUser() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
 
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
+    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(false);
+    loginServlet.setUserStore(mockUserStore);
+   
     loginServlet.doPost(mockRequest, mockResponse);
 
-    Mockito.verify(mockRequest)
-        .setAttribute("error", "Please enter only letters, numbers, and spaces.");
+    Mockito.verify(mockRequest).setAttribute("error", "That username was not found."); 
     Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
   }
 
   @Test
-  public void testDoPost_NewUser() throws IOException, ServletException {
+  public void testDoPost_ExistingUserWithIncorrectPassword() throws IOException, ServletException{
     Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("bad password");
 
     UserStore mockUserStore = Mockito.mock(UserStore.class);
-    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(false);
-    loginServlet.setUserStore(mockUserStore);
+    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(true);
+    
+    User inputUser = new User(UUID.randomUUID(), "test username", "test password", Instant.now());
```

Comment:
I would name this just `user` or maybe `storedUser` instead, since it's coming from the datastore, not input like the request.

Same for the case below.